drivers/counter/counter_ifx_cat1.c

```diff
@@ -0,0 +1,488 @@
+/*
+ * Copyright (c) 2022 Cypress Semiconductor Corporation (an Infineon company) or
+ * an affiliate of Cypress Semiconductor Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief Counter driver for Infineon CAT1 MCU family.
+ */
+
+#define DT_DRV_COMPAT infineon_cat1_counter
+
+#include <zephyr/drivers/counter.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <cyhal_timer.h>
+
+#include <cyhal_tcpwm_common.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ifx_cat1_counter, CONFIG_COUNTER_LOG_LEVEL);
+
+/* Device config structure */
+struct ifx_cat1_counter_config {
+	struct counter_config_info counter_info;
+	TCPWM_CNT_Type *reg_addr;
+	uint32_t irqn;
+	uint8_t irq_priority;
+};
+
+/* Data structure */
+struct ifx_cat1_counter_data {
+	cyhal_timer_t counter_obj;
+	cyhal_timer_cfg_t counter_cfg;
+	struct counter_alarm_cfg alarm_cfg_counter;
+	struct counter_top_cfg top_value_cfg_counter;
+	uint32_t guard_period;
+	cyhal_resource_inst_t hw_resource;
+	bool alarm_irq_flag;
+};
+
+/* Default Counter configuration structure */
+const cy_stc_tcpwm_counter_config_t _cyhal_timer_default_config = {
```

Comment:
Drop the `_` prefix, we don't use it in Zephyr. Also make it `static`.

Reply:
Updated